### PR TITLE
Adding Fibre Channel lib

### PR DESCRIPTION
**Pull request by mathu97**
Fibre Channel lib initial commit

### CODE REVIEW

**Comment by mathu97** [comment]
@j-griffith @screeley44 @rootfs PTAL

**Comment by rootfs** [comment]
wouldn't apache license in every .go file?

**Comment by mathu97** [comment]
@rootfs, Added apache license

File: fibrechannel/fibrechannel.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+}
+
+func init() {
+	// TODO: add a handle to configure loggers after init
+	// also, make default for trace to go to discard when you're done messing around
+	log = logger.NewLogger(os.Stdout, os.Stdout, os.Stdout, os.Stderr)
+}
+
+// FindMultipathDeviceForDevice given a device name like /dev/sdx, find the devicemapper parent
+func FindMultipathDeviceForDevice(device string, io ioHandler) string {
```
Comment (rootfs) [inline fibrechannel/fibrechannel.go]: return the error
```suggestion
func FindMultipathDeviceForDevice(device string, io ioHandler) (string, error) {
```

File: fibrechannel/fibrechannel.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+// given a wwn and lun, find the device and associated devicemapper parent
+func findDisk(wwn, lun string, io ioHandler) (string, string) {
+	FcPath := "-fc-0x" + wwn + "-lun-" + lun
+	DevPath := "/dev/disk/by-path/"
+	if dirs, err := io.ReadDir(DevPath); err == nil {
+		for _, f := range dirs {
+			name := f.Name()
+			if strings.Contains(name, FcPath) {
+
```
Comment (rootfs) [inline fibrechannel/fibrechannel.go]: remove

File: fibrechannel/fibrechannel.go [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+				return disk, dm
+			}
+		}
+	}
+	log.Error.Printf("fc: failed to find a disk [%s]", DevID+FcPath)
+	return "", ""
+}
+
+// Connect attempts to connect a fc volume to a node using the provided Connector info
+func Connect(c Connector, io ioHandler) (string, error) {
```
Comment (rootfs) [inline fibrechannel/fibrechannel.go]: Connect -> Attach (as frequently used in kubernetes volumes)

File: fibrechannel/fibrechannel.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+	if err != nil {
+		log.Error.Printf("unable to find disk given WWNN or WWIDs")
+		return "", err
+	}
+
+	return devicePath, nil
+}
+
+// Disconnect performs a disconnect operation on a volume
+func Disconnect(devicePath string, io ioHandler) error {
```
Comment (rootfs) [inline fibrechannel/fibrechannel.go]: Disconnect -> Detach

File: logger/logger.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package logger
```
Comment (rootfs) [inline logger/logger.go]: why reimplement logger, glog can print trace, warning, and error and can adjust verbosity.

Comment (mathu97) [reply logger/logger.go]: Changed it to glog instead

**Pull request merged**